src/serializer/visitors.js

```diff
@@ -251,9 +251,9 @@ export let ClosureRefVisitor = {
     if (ignorePath(path)) return;
 
     let innerName = path.node.name;
-    // TODO #1621: doesn't check that `arguments` is in top function.
     if (innerName === "arguments") {
-      state.functionInfo.usesArguments = true;
+      state.functionInfo.usesArguments = state.functionInfo.depth === 1;
```

Comment:
That doesn't look right, as it will reset `usesArguments` to false depending on the order in which it visits.

Reply:
Oh, you're right!